src/Workspaces/Core/Desktop/Workspace/Storage/PersistentStorageService.cs

```diff
@@ -58,8 +58,11 @@ protected AbstractPersistentStorageService(IOptionService optionService, bool te
         protected abstract bool ShouldDeleteDatabase(Exception exception);
 
         public IPersistentStorage GetStorage(Solution solution)
+            => GetStorage(solution, checkBranchId: true);
+
+        public IPersistentStorage GetStorage(Solution solution, bool checkBranchId)
```

Comment:
I can't think of any names i like.  At the end of the day this is specifically aboutif we have the branch-id check, and nothing else.  So i felt the name was appropriate.